simpegEM/FDEM/FDEM.py

```diff
@@ -247,59 +350,59 @@ def getRHS(self, freq):
-                    SRCy = src(tx.loc, self.mesh.gridFy, 'y')
-                    SRCz = src(tx.loc, self.mesh.gridFz, 'z')
-
-                elif tx.txType == 'CircularLoop':
-                    src = Sources.MagneticLoopVectorPotential                
-                    SRCx = src(tx.loc, self.mesh.gridEx, 'x', tx.radius)
-                    SRCy = src(tx.loc, self.mesh.gridEy, 'y', tx.radius)
-                    SRCz = src(tx.loc, self.mesh.gridEz, 'z', tx.radius)
-                else:
-
-                    raise NotImplemented('%s txType is not implemented' % tx.txType)
-                SRC = np.concatenate((SRCx, SRCy, SRCz))                                    
-
-            else:
-                raise Exception('Unknown mesh for VMD')           
+        #     elif self.mesh._meshType is 'TENSOR':
+
+        #         if tx.txType == 'VMD':
+        #             src = Sources.MagneticDipoleVectorPotential
+        #             SRCx = src(tx.loc, self.mesh.gridEx, 'x')
+        #             SRCy = src(tx.loc, self.mesh.gridEy, 'y')
+        #             SRCz = src(tx.loc, self.mesh.gridEz, 'z')
+
+        #         elif tx.txType == 'VMD_B':
+        #             src = Sources.MagneticDipoleFields
+        #             SRCx = src(tx.loc, self.mesh.gridFx, 'x')
+        #             SRCy = src(tx.loc, self.mesh.gridFy, 'y')
+        #             SRCz = src(tx.loc, self.mesh.gridFz, 'z')
+
+        #         elif tx.txType == 'CircularLoop':
+        #             src = Sources.MagneticLoopVectorPotential                
+        #             SRCx = src(tx.loc, self.mesh.gridEx, 'x', tx.radius)
+        #             SRCy = src(tx.loc, self.mesh.gridEy, 'y', tx.radius)
+        #             SRCz = src(tx.loc, self.mesh.gridEz, 'z', tx.radius)
+        #         else:
+
+        #             raise NotImplemented('%s txType is not implemented' % tx.txType)
+        #         SRC = np.concatenate((SRCx, SRCy, SRCz))                                    
+
+        #     else:
+        #         raise Exception('Unknown mesh for VMD')           
             
-            rhs[i] = SRC
+        #     rhs[i] = SRC
             
-        mui = self.MfMui            
-        if tx.txType == 'VMD_B':
-            b_0 = np.concatenate(rhs).reshape((self.mesh.nF, len(Txs)), order='F')
-        else:            
-            a = np.concatenate(rhs).reshape((self.mesh.nE, len(Txs)), order='F')
-            C = self.mesh.edgeCurl
-            b_0 = C*a
+        # mui = self.MfMui            
+        # if tx.txType == 'VMD_B':
+        #     b_0 = np.concatenate(rhs).reshape((self.mesh.nF, len(Txs)), order='F')
+        # else:            
+        #     a = np.concatenate(rhs).reshape((self.mesh.nE, len(Txs)), order='F')
+        #     C = self.mesh.edgeCurl
+        b_0 = getSource(self,freq) #C*a
 
-        return -1j*omega(freq)*mui*b_0
+        return -1j*omega(freq)*b_0
```

Comment:
In the isotropic case, yes. Anisotropic, octree, or LRM you are quite right that we need to multiply by the transpose of the mass matrix in all cases. i.e. the entire equation by mfmui * (   ...  = src )